packages/material-ui/src/InputBase/inputBaseClasses.ts

```diff
@@ -17,6 +17,8 @@ export interface InputBaseClasses {
   error: string;
   /** Styles applied to the input element if `size="small"`. */
   sizeSmall: string;
+  /** Styles applied to the input element if `size="medium"`. */
+  sizeMedium: string;
```

Comment:
`sizeMedium` and `inputSizeMedium` are never applied on the DOM. https://github.com/mui-org/material-ui/blob/8cc7dd0568adde2b8c23804fabdadbd7cfff21d6/packages/material-ui/src/InputBase/InputBase.js#L28 . In #25969 I was showcasing the problem with adding too many classes on the DOM nodes. I would personally advocate removing the default class names, to match Bootstrap: https://getbootstrap.com/docs/5.0/components/buttons/#sizes.

Reply:
Good catch, in my opinion, we should add those classes. Again I think we are coming back to this too many times :) On the PR that you linked, `margin: none` does not provide any styles, and it probably should never provide, but with the sizing props is different, especially if developers change the default props. If the default size for the component is changed to be `small`, there is no way of how the component can be customized in case size="medium"

Reply:
> especially if developers change the default props.

I would advocate that they shouldn't, we shouldn't make it easier for them. Instead, they can customize all the values to shift the values.

Reply:
Ok, I probably have hard time making a call of when this is ok and when not. Should we make it consistent for all props in that case? What I am trying to understand is, is this one-time thing, or should we follow this thinking everywhere. For example, if in a component, the `primary` is the default color, should we provide a class for it? I think it would be best if we have a rule that applies everwhere, thank having this discussion on each change we do :) (it's my fault 100%, but I am trying to make us more productive :D)

Reply:
For the color, I would come back to what I was saying around following Bootstrap. Being primary seems subjective, it makes sense to have a class name/style override.

Reply:
Agree, and following Bootstrap as a base seems like a great start.

Reply:
Agree, we should remove all "medium" size if it means "default". will revert the change.

Reply:
related issue about `default`. https://github.com/mui-org/material-ui/issues/13028